### PR TITLE
Better media thumbnails including WebP support

### PR DESCRIPTION
**Pull request by IKarbowiak**
- Use `ImageField` instead of `VersatileImageField` in `Account`, `Category`, `Collection`, `ProductMedia` models
- Introduce `Thumbnail` model that keeps the thumbnail images of the account avatar, category and collection background images, and product media images
- Do not create pre-create thumbnails. When somebody asks for an image, look for when existing `Thumbnail` object - if exists return the thumbnail image URL, otherwise return the proxy URL. When somebody opens the proxy URL the thumbnail is created and returned.
- Limit number of available sizes: 32, 64, 128, 256, 512, 1024, 2048, 4096
- Alow specifying the image size and format when asking for image field in API
- Get rid of versatile settings.

Docs: https://github.com/saleor/saleor-docs/pull/463

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6823 Drop django-versatileimagefield](https://app.clickup.com/t/2549495/SALEOR-6823)

**Comment by IKarbowiak** [comment]
The migrations check is failing, but it will be released in 3.5 , so it's no problem.

**Review by korycins — APPROVED** [review]

File: saleor/thumbnail/__init__.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+default_app_config = "saleor.thumbnail.app.ThumbnailAppConfig"
+
+# defines the available thumbnail resolutions
+THUMBNAIL_SIZES = [32, 64, 128, 256, 512, 1024, 2048, 4096]
+
+
+class ThumbnailFormat:
+    # ORIGINAL = "original"
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/thumbnail/__init__.py:8]: No. Thanks!

File: saleor/graphql/account/types.py [hunk 2]
```diff
@@ -286,7 +290,24 @@ class User(ModelObjectType):
...
+        Image,
+        size=graphene.Int(
+            description=(
+                "Size of the avatar. If not provided the original image "
+                "will be returned."
+            )
+        ),
+        format=ThumbnailFormatEnum(
+            description=(
+                "The format of the avatar. When not provided format of the original "
```
Comment (maarcingebala) [inline saleor/graphql/account/types.py:303]:
```suggestion
                "The format of the avatar. When not provided, format of the original "
```

File: saleor/graphql/account/types.py [hunk 3]
```diff
@@ -430,15 +451,29 @@ def _resolve_orders(orders):
...
+            return Image(url=root.avatar.url, alt=None)
+
+        format = format.lower() if format else None
+        size = get_thumbnail_size(size)
+
+        def _resolve_avatar(thumbnail):
+            url = (
+                prepare_image_proxy_url(root.id, "User", size, format)
+                if thumbnail is None
+                else thumbnail.image.url
```
Comment (maarcingebala) [inline saleor/graphql/account/types.py:468]: Maybe we could have a function that encapsulates exactly this - returns proxy URL or thumbnail URL it's ready?

File: saleor/graphql/order/types.py [hunk 4]
```diff
@@ -523,7 +525,22 @@ class OrderLine(ModelObjectType):
     thumbnail = graphene.Field(
         Image,
         description="The main thumbnail for the ordered product.",
-        size=graphene.Argument(graphene.Int, description="Size of thumbnail."),
+        size=graphene.Argument(
```
Comment (maarcingebala) [inline saleor/graphql/order/types.py:528]: Definition of the `thumbnail` field seems the same in all cases, I was wondering if it wouldn't be nice to define a `ThumbnailField` that extends the `graphene.Field` and predefines the `size` and `format` arguments with these descriptions? Not sure how easy/hard it is to add, but might be worth considering.

File: saleor/core/tasks.py [hunk 5]
```diff
@@ -40,6 +40,6 @@ def delete_event_payloads_task():
 def delete_product_media_task(media_id):
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/core/tasks.py:40]: Internat task: [SALEOR-7604](https://app.clickup.com/t/2549495/SALEOR-7604)

**Review by fowczarek — APPROVED** [review]

**Pull request merged**